src/main/java/com/eprosima/fastdds/idl/templates/TypesCdrAuxHeaderImpl.stg

```diff
@@ -0,0 +1,463 @@
+// Copyright 2023 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+group TypesCdrAuxHeader;
+
+import "eprosima.stg"
+
+main(ctx, definitions, extensions) ::= <<
+$fileHeader(ctx=ctx, file=[ctx.filename, "CdrAux.ipp"], description=["This source file contains some declarations of CDR related functions."])$
+
+#ifndef _FAST_DDS_GENERATED_$ctx.headerGuardName$CDRAUX_IPP_
+#define _FAST_DDS_GENERATED_$ctx.headerGuardName$CDRAUX_IPP_
+
+#include "$ctx.filename$CdrAux.hpp"
+
+$if(ctx.cdr)$
+#include <fastcdr/Cdr.h>
+#include <fastcdr/CdrSizeCalculator.hpp>
+
+$endif$
+
+#include <fastcdr/exceptions/BadParamException.h>
+using namespace eprosima::fastcdr::exception;
+
+namespace eprosima {
+namespace fastcdr {
+
+$definitions; separator="\n"$
+
+} // namespace fastcdr
+} // namespace eprosima
+
+#endif // _FAST_DDS_GENERATED_$ctx.headerGuardName$CDRAUX_IPP_
+>>
+
+module(ctx, parent, module, definition_list) ::= <<
+$definition_list$
+>>
+
+definition_list(definitions) ::= <<
+$definitions; separator="\n"$
+
+>>
+
+annotation(ctx, annotation) ::= <<>>
+
+interface(ctx, parent, interface, export_list) ::= <<
+$export_list$
+>>
+
+export_list(exports) ::= <<
+$exports; separator="\n"$
+
+>>
+
+fwd_decl(ctx, parent, type) ::= <<>>
+
+exception(ctx, parent, exception) ::= <<
```

Comment:
As far as I know the exception definition for IDLs is not officially supported. I suggest removing this code which is not tested nor used.